File: backend/CustomerUpdate.API/DTOs/CustomerDto.cs
using System.ComponentModel.DataAnnotations;

namespace CustomerUpdate.API.DTOs;

public class CustomerDto
{
    public int CustomerId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public string? Email { get; set; }
    public string Address { get; set; } = string.Empty;
    public string TaxId { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public string Status { get; set; } = "Active";
}

public class CreateCustomerDto
{
    [Required]
    [StringLength(100, MinimumLength = 2)]
    public string Name { get; set; } = string.Empty;
    
    [Phone]
    public string? Phone { get; set; }
    
    [EmailAddress]
    public string? Email { get; set; }
    
    [Required]
    [StringLength(500, MinimumLength = 5)]
    public string Address { get; set; } = string.Empty;
    
    [Required]
    [StringLength(11, MinimumLength = 11)]
    [RegularExpression(@"^\d{11}$", ErrorMessage = "Tax ID must be exactly 11 digits")]
    public string TaxId { get; set; } = string.Empty;
}

public class UpdateCustomerDto
{
    [Required]
    [StringLength(100, MinimumLength = 2)]
    public string Name { get; set; } = string.Empty;
    
    [Phone]
    public string? Phone { get; set; }
    
    [EmailAddress]
    public string? Email { get; set; }
    
    [Required]
    [StringLength(500, MinimumLength = 5)]
    public string Address { get; set; } = string.Empty;
    
    [Required]
    [StringLength(11, MinimumLength = 11)]
    [RegularExpression(@"^\d{11}$", ErrorMessage = "Tax ID must be exactly 11 digits")]
    public string TaxId { get; set; } = string.Empty;
}

public class ApiResponse<T>
{
    public bool Success { get; set; }
    public string Message { get; set; } = string.Empty;
    public T? Data { get; set; }
    public List<string>? Errors { get; set; }
    public string? Error { get; set; }
} 

File: backend-java/src/main/java/com/example/customerupdate/dto/CustomerDto.java
package com.example.customerupdate.dto;

import jakarta.validation.constraints.*;
import java.time.LocalDateTime;
import java.util.List;

public class CustomerDto {
    private Integer customerId;
    private String name;
    private String phone;
    private String email;
    private String address;
    private String taxId;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String status;
    // Getters and setters omitted for brevity
}

class CreateCustomerDto {
    @NotBlank
    @Size(min = 2, max = 100)
    private String name;
    @Pattern(regexp = "^$|^[+]?[0-9\- ]{7,20}$", message = "Invalid phone number")
    private String phone;
    @Email
    private String email;
    @NotBlank
    @Size(min = 5, max = 500)
    private String address;
    @NotBlank
    @Size(min = 11, max = 11)
    @Pattern(regexp = "^\\d{11}$", message = "Tax ID must be exactly 11 digits")
    private String taxId;
    // Getters and setters omitted for brevity
}

class UpdateCustomerDto {
    @NotBlank
    @Size(min = 2, max = 100)
    private String name;
    @Pattern(regexp = "^$|^[+]?[0-9\- ]{7,20}$", message = "Invalid phone number")
    private String phone;
    @Email
    private String email;
    @NotBlank
    @Size(min = 5, max = 500)
    private String address;
    @NotBlank
    @Size(min = 11, max = 11)
    @Pattern(regexp = "^\\d{11}$", message = "Tax ID must be exactly 11 digits")
    private String taxId;
    // Getters and setters omitted for brevity
}

class ApiResponse<T> {
    private boolean success;
    private String message;
    private T data;
    private List<String> errors;
    private String error;
    // Getters and setters omitted for brevity
} 